enhancements/testing/performance/README.md

```diff
@@ -0,0 +1,189 @@
+3. **Garbage collection validation** – Confirm that cleanup of child resources when a Project is deleted is timely and does not leave orphaned objects.
+
+We will explore two frameworks to drive this testing:
+
+- **[ClusterLoader2 (CL2)](https://github.com/kubernetes/perf-tests/tree/master/clusterloader2)**: the upstream Kubernetes scalability and performance testing tool. It provides YAML-driven test definitions, standard measurements (API responsiveness, etcd metrics, resource usage), and integration with Prometheus.  
+- **[k6](https://k6.io/)**: a modern load testing tool with a JavaScript-based DSL. It gives us flexibility for defining custom scenarios and is well-suited to exercising API endpoints directly, including Milo’s virtual API server paths.
+
+The goal is to evaluate both, run representative workloads with each, and decide which is the better long-term fit for Milo’s perf testing.
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this Enhancement.  Describe why the change is important and the benefits to users.
+-->
+
+Milo is intended to host many independent tenants through Projects. Each Project runs its own virtual API server surface area but shares the same backing etcd and controller stack. Before pushing toward production scale, we need data to answer:
+
+- **How many Projects can Milo support concurrently?**  
+- **How many Secrets/ConfigMaps per Project before etcd or API server usage becomes problematic?**  
+- **What is the performance profile during resource churn (create → steady-state → delete)?**  
+- **Does garbage collection of Project resources complete reliably under load?**
+
+Without this testing, we risk regressions in latency, uncontrolled memory growth, and operational issues when cleaning up Projects.
+
+
+### Goals
+
+<!--
+List the specific goals of the Enhancement. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Automate creation of N Projects and high-volume Secrets/ConfigMaps workloads inside each Project.  
+- Exercise deletion flows and verify cleanup of Project-scoped resources.  
+- Collect metrics on API server responsiveness, availability, and etcd performance.  
+- Provide repeatable developer workflows via Taskfile targets to run tests locally (kind) or against remote clusters (e.g., GKE).  
+- Compare CL2 vs k6 in terms of:
+  - Ease of expressing tests.  
+  - Observability and built-in measurements.  
+  - Ability to scale workloads.  
+  - Integration into CI pipelines.  
+
+### Non-Goals
+
+<!--
+What is out of scope for this Enhancement? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Testing every type of Kubernetes object (we’ll focus on Projects + Secrets/ConfigMaps).  
+- Full multi-tenant end-to-end validation (authn/authz, networking, etc.).  
```

Comment:
@zachsmith1 by this do you mean perf testing won't confirm that authn/authz works as expected? 

I do think there's value in putting authn/authz under load to see how it handles large scale deployments with a large number of organizations, projects, and role bindings, but would agree that's likely separate from testing the multi-tenant aspect of the control plane.

Reply:
Ya maybe "validation" wasn't the right word. It might make sense to include some metrics around authn/authz under load here within this scope. We can build upon this later if needed